### DISCUSSION TITLE
How to use the existing  PNP Powershell commands with Sites.Selected Permission in Azure AD

### DISCUSSION THREAD

**Discussion by in8awaka** [pnp/powershell#1476] [Q&A]
Hi All,

I am trying to Read / Write data in SP site using PNP.PowerShell with and Azure AD App having Sites.Selected Permission. I generated the token using client id and certificate and get the data using Graph API (Invoke-WebRequest) method 

When i use the same token in PNP.Powershell using 
Connect-PnPOnline -Url $siteURL -AccessToken $accessToken

it connected but not giving any result in Get-PnpList it gives me 401 . 

so wanted to know if it is planned in future releases for Graph API token will work in PNP.Powershell commands also.

![image](https://user-images.githubusercontent.com/49389070/147533693-d3096d82-e9e3-4ac9-834a-c447e32e9024.png)

**Comment by LeonArmston** [comment]
Answered in https://github.com/pnp/powershell/discussions/1480
